test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2353,10 +2353,25 @@ class test_ucp_sockaddr_protocols_diff_config : public test_ucp_sockaddr_protoco
         }
 
         m_err_count = 0;
+        modify_config("CM_USE_ALL_DEVICES", cm_use_all_devices() ? "y" : "n");
+
         get_sockaddr();
         test_base::init();
     }
 
+    static void
+    get_test_variants(std::vector<ucp_test_variant>& variants) {
+        uint64_t features = UCP_FEATURE_TAG | UCP_FEATURE_STREAM |
+                            UCP_FEATURE_RMA | UCP_FEATURE_AM;
+
+        add_variant_with_value(variants, features, TEST_MODIFIER_MT |
+                               TEST_MODIFIER_CM_USE_ALL_DEVICES,
+                               "mt,all_devs", MULTI_THREAD_WORKER);
+
+        add_variant_with_value(variants, features, TEST_MODIFIER_MT,
+                               "mt,not_all_devs", MULTI_THREAD_WORKER);
```

Comment:
why it tests only MT case?
I'd expect both MT and npn-MT variants should be added here.
pls, use `get_test_variants_mt()` for that purpose

Reply:
IMO for NUM_PATHS=1 vs. 2 no need to test both MT/non-MT

Reply:
> IMO for NUM_PATHS=1 vs. 2 no need to test both MT/non-MT

should it test non-MT only cases then?

Reply:
yes, better remove MT then @changchengx

Reply:
[class test_ucp_sockaddr_protocols_diff_config](https://github.com/openucx/ucx/blob/v1.11.2/test/gtest/ucp/test_ucp_sockaddr.cc#L2001) is derived from [class test_ucp_sockaddr_protocols](https://github.com/openucx/ucx/blob/v1.11.2/test/gtest/ucp/test_ucp_sockaddr.cc#L1459). The [test_ucp_sockaddr_protocols::get_test_variants](https://github.com/openucx/ucx/blob/v1.11.2/test/gtest/ucp/test_ucp_sockaddr.cc#L1470) is specified to do test under multi-thread when creating ucp_worker, should we also keep the same parameter here to do test under multi-thread when creating ucp_worker in ```class test_ucp_sockaddr_protocols_diff_config``` ? If not, I'll change this PR to test for non-MT cases.

Reply:
I've decided to change it to be non-MT cases.

Reply:
@yosefe MT is removed